exporter/collector/internal/datapointstorage/datapointcache.go

```diff
@@ -71,118 +77,130 @@ func NewCache(shutdown <-chan struct{}) Cache {
 
 // GetNumberDataPoint retrieves the point associated with the identifier, and whether
 // or not it was found
-func (c Cache) GetNumberDataPoint(identifier string) (*pmetric.NumberDataPoint, bool) {
+func (c *Cache) GetNumberDataPoint(identifier string) (*pmetric.NumberDataPoint, bool) {
+	c.numberLock.RLock()
+	defer c.numberLock.RUnlock()
 	point, found := c.numberCache[identifier]
 	if found {
-		point.used = true
+		point.used.Store(true)
 		c.numberCache[identifier] = point
 	}
 	return point.point, found
 }
 
 // SetNumberDataPoint assigns the point to the identifier in the cache
-func (c Cache) SetNumberDataPoint(identifier string, point *pmetric.NumberDataPoint) {
-	c.numberCache[identifier] = usedNumberPoint{point, true}
+func (c *Cache) SetNumberDataPoint(identifier string, point *pmetric.NumberDataPoint) {
+	c.numberLock.Lock()
+	defer c.numberLock.Unlock()
+	c.numberCache[identifier] = usedNumberPoint{point, atomic.NewBool(true)}
 }
 
 // GetSummaryDataPoint retrieves the point associated with the identifier, and whether
 // or not it was found
-func (c Cache) GetSummaryDataPoint(identifier string) (*pmetric.SummaryDataPoint, bool) {
+func (c *Cache) GetSummaryDataPoint(identifier string) (*pmetric.SummaryDataPoint, bool) {
+	c.summaryLock.RLock()
+	defer c.summaryLock.RUnlock()
 	point, found := c.summaryCache[identifier]
 	if found {
-		point.used = true
+		point.used.Store(true)
 		c.summaryCache[identifier] = point
 	}
 	return point.point, found
 }
 
 // SetSummaryDataPoint assigns the point to the identifier in the cache
-func (c Cache) SetSummaryDataPoint(identifier string, point *pmetric.SummaryDataPoint) {
-	c.summaryCache[identifier] = usedSummaryPoint{point, true}
+func (c *Cache) SetSummaryDataPoint(identifier string, point *pmetric.SummaryDataPoint) {
+	c.summaryLock.Lock()
+	defer c.summaryLock.Unlock()
+	c.summaryCache[identifier] = usedSummaryPoint{point, atomic.NewBool(true)}
 }
 
 // GetHistogramDataPoint retrieves the point associated with the identifier, and whether
 // or not it was found
-func (c Cache) GetHistogramDataPoint(identifier string) (*pmetric.HistogramDataPoint, bool) {
+func (c *Cache) GetHistogramDataPoint(identifier string) (*pmetric.HistogramDataPoint, bool) {
+	c.histogramLock.RLock()
+	defer c.histogramLock.RUnlock()
 	point, found := c.histogramCache[identifier]
 	if found {
-		point.used = true
+		point.used.Store(true)
 		c.histogramCache[identifier] = point
```

Comment:
Same concern here.   You're writing to histogram cache without a write lock, but I'm not sure you need to write to it.